modules/blockuser.cpp

```diff
@@ -173,21 +181,22 @@ class CBlockUser : public CModule {
 
 		// Disconnect all clients
 		vector<CClient*> vpClients = pUser->GetAllClients();
-		vector<CClient*>::iterator it;
-		for (it = vpClients.begin(); it != vpClients.end(); ++it) {
-			(*it)->PutStatusNotice(MESSAGE);
-			(*it)->Close(Csock::CLT_AFTERWRITE);
+		for (CClient* pClient : vpClients) {
+			pClient->PutStatusNotice("Blocked: " + (sReason.empty() ? DEFAULT_REASON : sReason));
+			pClient->Close(Csock::CLT_AFTERWRITE);
 		}
 
 		// Disconnect all networks from irc
 		vector<CIRCNetwork*> vNetworks = pUser->GetNetworks();
-		for (vector<CIRCNetwork*>::iterator it2 = vNetworks.begin(); it2 != vNetworks.end(); ++it2) {
-			(*it2)->SetIRCConnectEnabled(false);
+		for (CIRCNetwork* pNetwork : vNetworks) {
+			pNetwork->SetIRCConnectEnabled(false);
 		}
 
-		SetNV(pUser->GetUserName(), "");
+		SetNV(pUser->GetUserName(), sReason);
```

Comment:
Well, I'm spoiled by protobuf, but that certainly won't be a good dependency for ZNC.
`SetNV(pUser->GetUserName() + "/reason", sReason)` would make it harder to use, and to e.g. iterate over NV like it does now.
Probably MCString can be extended to provide [un]serialization of map? It already does that when writing NV to disk.
Or json?